pkg/apis/servicecatalog/types.go

```diff
@@ -439,3 +439,15 @@ const (
 const (
 	FinalizerServiceCatalog string = "kubernetes-incubator/service-catalog"
 )
+
+// Parameters is a union type for supporting different ways of representing parameters used for:
+// - Instance input parameters
+// - Binding output credentials
+type Parameters struct {
+	// Reference to a secret in the namespace.
```

Comment:
This removes the ability to do inline parameters -- which I suspect might break some users.  Should we include the ability to use raw (maybe call it 'embedded' or something like that) with a note that secret parameters should come from a secret?